### PR TITLE
Additional SM Drivers for CephFS and XOSAN

### PR DESCRIPTION
**Pull request by rushikeshjadhav**

### CODE REVIEW

File: XOSANSR.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
+#!/usr/bin/python
+#
+#
+# XOSANSR: Xen Orchestra SAN filesystem based storage repository
```
Comment (olivierlambert) [inline XOSANSR.py:4]: I think we should make some modifications here:

* we are introducing a generic Gluster driver so the name should be related to that (Gluster driver)
* this code is Open Source unlike what's in this header

Comment (rushikeshjadhav) [reply XOSANSR.py:4]: Ok, let's keep common header for both SR drivers. @nraynaud Can you do the modifications?

**Comment by stormi** [comment]
With my packager hat on: what are the external dependencies for the CephFS driver? I know those from the GlusterFS driver, I think.

To avoid installing all dependencies on hosts when their users only want one of those drivers, I may create separate RPMs, one per driver, each with its own dependencies. This does not impact this pull request in itself: I'm just preparing the next steps.

**Comment by rushikeshjadhav** [comment]
Following are the dependency 
```
# yum install centos-release-ceph-nautilus --enablerepo=extras
# yum install ceph-common
```

Actually we only need `ceph-common`, do you support importing packages from foreign repo?

**Comment by stormi** [comment]
Hmm, ok so those are dependencies from another repository. Ideally, we'd build it on our own buildsystem to have it available in our repositories. If this is difficult or too much work or not a priority for us (yet?), then I'd just document how to install the required dependencies and provide the driver without hard RPM dependencies.

**Comment by stormi** [comment]
How does the driver behave if ceph is not installed? Does it fail gracefully?

**Comment by rushikeshjadhav** [comment]
That would be ok to let user follow documentation. This will keep drivers together without having to create separate RPMs. It's just extra work not difficult. 

If it's unable to mount the mountpoint, the SR creation will fail giving error "mount failed with return code" or similar. Line 152

**Pull request closed without merging**